src/core/execute.c

```diff
@@ -2300,7 +2295,8 @@ static int exec_child(
         const char *home = NULL, *shell = NULL;
         dev_t journal_stream_dev = 0;
         ino_t journal_stream_ino = 0;
-        bool needs_mount_namespace;
+        bool needs_exec_restrictions, needs_mount_namespace,
+                needs_selinux = false, needs_smack = false, needs_apparmor = false;
```

Comment:
It seems that `needs_*` should be enclosed in `#ifdef`s so as not to cause something like the following:
```sh
../src/core/execute.c: In function ‘exec_child’:
../src/core/execute.c:2299:61: warning: unused variable ‘needs_apparmor’ [-Wunused-variable]
                 needs_selinux = false, needs_smack = false, needs_apparmor = false;
                                                             ^~~~~~~~~~~~~~
```